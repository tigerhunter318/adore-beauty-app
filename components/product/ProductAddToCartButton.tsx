import React from 'react'
import { StyleSheet } from 'react-native'
import { isSmallDevice } from '../../utils/device'
import Container from '../ui/Container'
import Type from '../ui/Type'
import theme from '../../constants/theme'
import AdoreSvgIcon from '../ui/AdoreSvgIcon'
import CatalogProductProvider from './CatalogProductProvider'
import OutOfStockSubscribe from './OutOfStockSubscribe'
import Icon from '../ui/Icon'
import formatProductInventory from './utils/formatProductInventory'
import { logObjectToJson } from '../../utils/logInfo'

const styles = StyleSheet.create({
  buttonContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 40,
    borderRadius: 3,
    flexDirection: 'row'
  },
  button: {
    textTransform: 'uppercase'
  },
  label: {
    paddingLeft: 10,
    fontSize: 12,
    letterSpacing: 1,
    textTransform: 'uppercase'
  },
  backordersContainer: {
    borderWidth: 1.2,
    borderColor: theme.orange
  },
  backordersBadge: {
    position: 'absolute',
    top: -17,
    right: 4
  },
  soldOutLabel: {
    paddingLeft: 0,
    color: theme.textGreyDark
  }
})

const DefaultAddButton = ({
  onPress,
  disabled,
  isListingView,
  isCartLineItem
}: {
  onPress: () => void
  disabled: boolean
  isListingView: boolean
  isCartLineItem: boolean
}) => {
  const getLabel = () => {
    let label = isListingView ? 'Add' : 'Add to bag'
    if (isCartLineItem) {
      label = isListingView ? 'Added' : 'Added to bag'
    }
    if (disabled) {
      label = isListingView ? 'Add' : 'Adding...'
    }
    return label
  }

  return (
    <Container
      onPress={onPress}
      style={styles.buttonContainer}
      background={theme.orange}
      disabled={disabled}
      ph={isListingView ? 0 : 1}
      testID="Button.AddToCart"
    >
      <Type bold color={theme.white} size={12} style={styles.button} testID="Button.AddToCart.Type">
        {getLabel()}
      </Type>
    </Container>
  )
}

const GiftlistButton = ({ disabled }: { disabled: boolean }) => (
  <Container style={styles.buttonContainer} disabled={disabled}>
    <Icon color={theme.black} name="close" type="material" size={20} />
    <Type pl={0.5} semiBold style={styles.label} testID="GiftlistButton.Type">
      sold out
    </Type>
  </Container>
)

const WaitlistButton = ({
  disabled,
  productSku,
  productUrl,
  isModalVisible,
  onNotifyPress,
  isListingView
}: {
  disabled: boolean
  productSku: string
  productUrl: string
  isModalVisible: boolean
  onNotifyPress: () => void
  isListingView: boolean
}) => (
  <Container
    onPress={onNotifyPress}
    style={styles.buttonContainer}
    background={theme.lightGrey}
    disabled={disabled}
    ph={!isListingView ? 1 : 0}
  >
    <Container center rows>
      <AdoreSvgIcon name="email" height={14} width={16} />
      {!isListingView && !isSmallDevice() ? (
        <Type semiBold style={styles.label} testID="WaitlistButton.Type">
          notify me
        </Type>
      ) : null}
    </Container>
    <CatalogProductProvider productSku={productSku}>
      {({ catalogProductData }: any) => (
        <OutOfStockSubscribe
          isVisible={isModalVisible}
          onClose={onNotifyPress}
          baseVariantId={catalogProductData?.base_variant_id}
          id={catalogProductData?.id}
          productUrl={productUrl}
          isListingView={isListingView}
        />
      )}
    </CatalogProductProvider>
  </Container>
)

const BackordersButton = ({
  onPress,
  disabled,
  isListingView,
  isCarouselItem
}: {
  onPress: () => void
  disabled: boolean
  isListingView: boolean
  isCarouselItem?: boolean
}) => (
  <Container
    onPress={onPress}
    style={[styles.buttonContainer, styles.backordersContainer]}
    disabled={disabled}
    background={theme.white}
    ph={!isListingView ? 1 : 0}
  >
    {isListingView ? (
      <Container style={styles.backordersBadge}>
        <AdoreSvgIcon name="PreOrderBadge" color={theme.black} width={31} height={31} />
      </Container>
    ) : (
      <Container pr={0.5}>
        <AdoreSvgIcon name="PreOrder" width={30} height={30} />
      </Container>
    )}
    <Type semiBold style={styles.button} color={theme.orange} size={isSmallDevice() || isCarouselItem ? 8 : 12}>
      preorder
    </Type>
  </Container>
)

const SoldOutButton = ({ isListingView }: { isListingView: boolean }) => (
  <Container style={styles.buttonContainer} background={theme.lightGrey} ph={1}>
    <Type
      semiBold
      center
      style={[styles.label, styles.soldOutLabel, { fontSize: isListingView ? 8 : 12 }]}
      testID="SoldOutButton.Type"
    >
      sold out
    </Type>
  </Container>
)

type ProductAddToCartButtonProps = {
  productData: object
  productVariant: object
  isCartLineItem: boolean
  isModalVisible: boolean
  isListingView: boolean
  disabled: boolean
  productSku: string
  productUrl: string
  onNotifyPress: () => void
  onPress: () => void
  isCarouselItem?: boolean
}

const ProductAddToCartButton = ({ productData, productVariant, ...props }: ProductAddToCartButtonProps) => {
  const { isSoldOut, isOutOfStock, isBackordersOutOfStock, isGiftlistItemOutOfStock } = formatProductInventory(
    productData,
    productVariant
  )

  // console.log('198', '', 'ProductAddToCartButton', productData, productVariant, props)
  // logObjectToJson(productData)

  switch (true) {
    case isGiftlistItemOutOfStock:
      return <GiftlistButton {...props} />
    case isOutOfStock:
      return <WaitlistButton {...props} />
    case isBackordersOutOfStock:
      return <BackordersButton {...props} />
    case isSoldOut:
      return <SoldOutButton {...props} />
    default:
      return <DefaultAddButton {...props} />
  }
}

export default ProductAddToCartButton
