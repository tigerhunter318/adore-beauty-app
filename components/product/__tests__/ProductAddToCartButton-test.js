/* eslint-disable global-require */
import React from 'react'
import TestRenderer from 'react-test-renderer'
import ProductAddToCartButton from '../ProductAddToCartButton'

jest.mock('../OutOfStockSubscribe', () => 'OutOfStockSubscribe')

const cases = [
  {
    name: 'ArticleShopProductInStock',
    productData: require('./__mocks__/ArticleShopProductInStock.json'),
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },
  {
    name: 'ArticleShopProductOutOfStock',
    productData: require('./__mocks__/ArticleShopProductOutOfStock.json'),
    expectations: [{ testID: 'WaitlistButton.Type', value: 'notify me' }]
  },
  {
    name: 'ArticleShopProductSoldOut',
    productData: require('./__mocks__/ArticleShopProductSoldOut.json'),
    expectations: [{ testID: 'SoldOutButton.Type', value: 'sold out' }]
  },
  {
    name: 'GiftListItemInStock',
    productData: require('./__mocks__/GiftListItemInStock.json'),
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },
  {
    name: 'GiftListItemSoldOut',
    productData: require('./__mocks__/GiftListItemSoldOut.json'),
    expectations: [{ testID: 'GiftlistButton.Type', value: 'sold out' }]
  },
  {
    name: 'SearchProductCardInStock',
    productData: require('./__mocks__/SearchProductCardInStock.json'),
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },
  {
    name: 'SearchProductCardSoldOut',
    productData: require('./__mocks__/SearchProductCardSoldOut.json'),
    expectations: [{ testID: 'SoldOutButton.Type', value: 'sold out' }]
  },

  {
    name: 'SearchProductCardOutOfStock',
    productData: require('./__mocks__/SearchProductCardOutOfStock.json'),
    expectations: [{ testID: 'WaitlistButton.Type', value: 'notify me' }]
  },

  {
    name: 'ProductDetailWithVariants',
    status: 'In stock',
    productData: require('./__mocks__/ProductDetailWithVariants.json'),
    productVariant: 14527,
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },

  {
    name: 'ProductDetailWithVariants',
    status: 'Out of stock',
    productData: require('./__mocks__/ProductDetailWithVariants.json'),
    productVariant: 927,
    expectations: [{ testID: 'WaitlistButton.Type', value: 'notify me' }]
  },

  {
    name: 'ProductDetailInStock',
    productData: require('./__mocks__/ProductDetailInStock.json'),
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },

  {
    name: 'ProductDetailOutOfStock',
    productData: require('./__mocks__/ProductDetailOutOfStock.json'),
    expectations: [{ testID: 'WaitlistButton.Type', value: 'notify me' }]
  },

  {
    name: 'CategoryProductListItemWithVariantsInStock',
    productData: require('./__mocks__/CategoryProductListItemWithVariantsInStock.json'),
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },

  {
    name: 'CategoryProductListItemSoldOut',
    productData: require('./__mocks__/CategoryProductListItemSoldOut.json'),
    expectations: [{ testID: 'WaitlistButton.Type', value: 'notify me' }]
  },

  {
    name: 'WishlistInStock',
    productData: require('./__mocks__/WishlistInStock.json'),
    expectations: [{ testID: 'Button.AddToCart.Type', value: 'Add to bag' }]
  },

  {
    name: 'WishListSoldOut',
    productData: require('./__mocks__/WishListSoldOut.json'),
    expectations: [{ testID: 'WaitlistButton.Type', value: 'notify me' }]
  }
]

/*
jest --watch components/product/__tests__

 */
describe('ProductAddToCartButton', () => {
  cases.forEach(({ name, status = '', expectations, productData, productVariant = undefined, props = {} }) => {
    it(`${name} ${status}.`, () => {
      const render = TestRenderer.create(
        <ProductAddToCartButton {...props} productVariant={productVariant} productData={productData || {}} />
      )

      expect(render.toJSON()).toMatchSnapshot()

      expectations.forEach(({ testID, value }) => {
        expect(render.root.findByProps({ testID }).props.children).toBe(value)
      })
    })
  })
})
