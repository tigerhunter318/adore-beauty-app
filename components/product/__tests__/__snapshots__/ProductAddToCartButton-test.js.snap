// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductAddToCartButton ArticleShopProductInStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;

exports[`ProductAddToCartButton ArticleShopProductOutOfStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <AdoreSvgIcon
      height={14}
      name="email"
      width={16}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="WaitlistButton.Type"
    >
      notify me
    </Text>
  </View>
  <OutOfStockSubscribe />
</View>
`;

exports[`ProductAddToCartButton ArticleShopProductSoldOut . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "textAlign": "center",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
          Object {
            "color": "#777777",
            "paddingLeft": 0,
          },
          Object {
            "fontSize": 12,
          },
        ],
      ]
    }
    testID="SoldOutButton.Type"
  >
    sold out
  </Text>
</View>
`;

exports[`ProductAddToCartButton CategoryProductListItemSoldOut . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <AdoreSvgIcon
      height={14}
      name="email"
      width={16}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="WaitlistButton.Type"
    >
      notify me
    </Text>
  </View>
  <OutOfStockSubscribe />
</View>
`;

exports[`ProductAddToCartButton CategoryProductListItemWithVariantsInStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;

exports[`ProductAddToCartButton GiftListItemInStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;

exports[`ProductAddToCartButton GiftListItemSoldOut . 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <Icon
    color="#000000"
    name="close"
    size={20}
    type="material"
  />
  <Text
    allowFontScaling={false}
    pl={0.5}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "paddingLeft": 5,
        },
        Object {
          "fontSize": 12,
          "letterSpacing": 1,
          "paddingLeft": 10,
          "textTransform": "uppercase",
        },
      ]
    }
    testID="GiftlistButton.Type"
  >
    sold out
  </Text>
</View>
`;

exports[`ProductAddToCartButton ProductDetailInStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;

exports[`ProductAddToCartButton ProductDetailOutOfStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <AdoreSvgIcon
      height={14}
      name="email"
      width={16}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="WaitlistButton.Type"
    >
      notify me
    </Text>
  </View>
  <OutOfStockSubscribe />
</View>
`;

exports[`ProductAddToCartButton ProductDetailWithVariants In stock. 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;

exports[`ProductAddToCartButton ProductDetailWithVariants Out of stock. 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <AdoreSvgIcon
      height={14}
      name="email"
      width={16}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="WaitlistButton.Type"
    >
      notify me
    </Text>
  </View>
  <OutOfStockSubscribe />
</View>
`;

exports[`ProductAddToCartButton SearchProductCardInStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;

exports[`ProductAddToCartButton SearchProductCardOutOfStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <AdoreSvgIcon
      height={14}
      name="email"
      width={16}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="WaitlistButton.Type"
    >
      notify me
    </Text>
  </View>
  <OutOfStockSubscribe />
</View>
`;

exports[`ProductAddToCartButton SearchProductCardSoldOut . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Montserrat-SemiBold",
          "textAlign": "center",
        },
        Array [
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
          Object {
            "color": "#777777",
            "paddingLeft": 0,
          },
          Object {
            "fontSize": 12,
          },
        ],
      ]
    }
    testID="SoldOutButton.Type"
  >
    sold out
  </Text>
</View>
`;

exports[`ProductAddToCartButton WishListSoldOut . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f5f3f3",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <AdoreSvgIcon
      height={14}
      name="email"
      width={16}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "Montserrat-SemiBold",
          },
          Object {
            "fontSize": 12,
            "letterSpacing": 1,
            "paddingLeft": 10,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="WaitlistButton.Type"
    >
      notify me
    </Text>
  </View>
  <OutOfStockSubscribe />
</View>
`;

exports[`ProductAddToCartButton WishlistInStock . 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#D6775E",
        "paddingLeft": 10,
        "paddingRight": 10,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
  testID="Button.AddToCart"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 12,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
    testID="Button.AddToCart.Type"
  >
    Add to bag
  </Text>
</View>
`;
